src/dataset.rs

```diff
@@ -0,0 +1,43 @@
+
+use qrlew::{builder::With, sql};
+use qrlew_sarus::data_spec;
+use std::rc::Rc;
+use pyo3::{pyclass, pymethods};
+
+use crate::{relation, error};
+
+#[pyclass]
+pub struct Dataset(pub data_spec::Dataset);
+
+#[pymethods]
+impl Dataset {
+    #[new]
+    pub fn new(dataset: &str, schema: &str, size: &str) -> error::Result<Self> {
+        Ok(Dataset(data_spec::Dataset::parse_from_dataset_schema_size(
+            dataset, schema, size,
+        )?))
+    }
+
+    pub fn relations(&self) -> Vec<(Vec<String>, relation::Relation)> {
+        self.0
+            .relations()
+            .into_iter()
+            .map(|(i, r)| (i, relation::Relation(r)))
+            .collect()
+    }
+
+    pub fn sql(&self, query: &str) -> error::Result<relation::Relation> {
```

Comment:
A bit strange to call this `sql` here and `parse` in the `Relation` object I would call this method taking a string and returning a relation `sql` like in spark.